src/main/scala/mesosphere/marathon/core/deployment/impl/DeploymentManagerActor.scala

```diff
@@ -124,22 +125,29 @@ import scala.util.control.NonFatal
   */
 // format: ON
 class DeploymentManagerActor(
-    taskTracker: InstanceTracker,
-    killService: KillService,
-    launchQueue: LaunchQueue,
-    scheduler: SchedulerActions,
-    healthCheckManager: HealthCheckManager,
-    eventBus: EventStream,
-    readinessCheckExecutor: ReadinessCheckExecutor,
-    deploymentRepository: DeploymentRepository,
-    deploymentActorProps: (ActorRef, KillService, SchedulerActions, DeploymentPlan, InstanceTracker, LaunchQueue, HealthCheckManager, EventStream, ReadinessCheckExecutor) => Props = DeploymentActor.props)(implicit val mat: Materializer) extends Actor with StrictLogging {
+                              metrics: Metrics,
+                              taskTracker: InstanceTracker,
+                              killService: KillService,
+                              launchQueue: LaunchQueue,
+                              scheduler: SchedulerActions,
+                              healthCheckManager: HealthCheckManager,
+                              eventBus: EventStream,
+                              readinessCheckExecutor: ReadinessCheckExecutor,
+                              deploymentRepository: DeploymentRepository,
+                              deploymentActorProps: (ActorRef, KillService, SchedulerActions, DeploymentPlan, InstanceTracker, LaunchQueue, HealthCheckManager, EventStream, ReadinessCheckExecutor) => Props = DeploymentActor.props)(implicit val mat: Materializer) extends Actor with StrictLogging {
   import context.dispatcher
 
   val runningDeployments: mutable.Map[String, DeploymentInfo] = mutable.Map.empty
   val deploymentStatus: mutable.Map[String, DeploymentStepInfo] = mutable.Map.empty
 
-  private[this] val runningDeploymentsMetric = Metrics.minMaxCounter(ServiceMetric, getClass, "currentDeploymentCount")
-  private[this] val totalDeploymentsMetric = Metrics.minMaxCounter(ServiceMetric, getClass, "deploymentCount")
+  private[this] val oldRunningDeploymentsMetric =
+    metrics.deprecatedMinMaxCounter(ServiceMetric, getClass, "currentDeploymentCount")
+  private[this] val newRunningDeploymentsMetric =
+    metrics.gauge("deployments.active")
+  private[this] val oldTotalDeploymentsMetric =
+    metrics.deprecatedMinMaxCounter(ServiceMetric, getClass, "deploymentCount")
```

Comment:
I've been wondering... since we already have a layer of indirection in our metrics, why not have this layer receive a new name and an old name, and then dispatch to each underlying library accordingly?

Reply:
Okay :) I thought that might be the case. We'll have to do lots of editing either way when we remove the old metrics.